diff --git a/configure.ac b/configure.ac
index 469d5399a4..942928fdca 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1774,7 +1774,7 @@ then
 
 		AC_MSG_CHECKING([for python distutils])
 		python_result="`$PYTHON -c 'import distutils;' 2>&1`"
-		if test -z "$python_result" ; then
+		if test $? -eq 0; then
 			python_has_distutils="yes"
 		else
 			python_has_distutils="no"
@@ -1895,7 +1895,7 @@ then
 
 		AC_MSG_CHECKING([for python3 distutils])
 		python3_result="`$PYTHON3 -c 'import distutils;' 2>&1`"
-		if test -z "$python3_result" ; then
+		if test $? -eq 0 ; then
 			python3_has_distutils="yes"
 		else
 			python3_has_distutils="no"
diff --git a/debian/bootstrap.sh b/debian/bootstrap.sh
index fa40702938..82540a1e05 100755
--- a/debian/bootstrap.sh
+++ b/debian/bootstrap.sh
@@ -2,7 +2,7 @@
 ##### -*- mode:shell-script; indent-tabs-mode:nil; sh-basic-offset:2 -*-
 ##### Author: Travis Cross <tc@traviscross.com>
 
-codename="sid"
+codename="noble"
 modulelist_opt=""
 modules_add=""
 use_sysvinit=""
@@ -30,7 +30,7 @@ lang_dir="../conf/vanilla/lang"
 fs_description="FreeSWITCH is a scalable open source cross-platform telephony platform designed to route and interconnect popular communication protocols using audio, video, text or any other form of media."
 mod_build_depends="." mod_depends="." mod_recommends="." mod_suggests="."
 supported_debian_distros="wheezy jessie stretch buster bullseye bookworm sid"
-supported_ubuntu_distros="trusty utopic xenial"
+supported_ubuntu_distros="trusty utopic xenial noble"
 supported_distros="$supported_debian_distros $supported_ubuntu_distros"
 avoid_mods=(
   applications/mod_sms_flowroute
@@ -103,6 +103,31 @@ avoid_mods_xenial=(
   asr_tts/mod_flite
   loggers/mod_raven
 )
+avoid_mods_noble=(
+  formats/mod_imagick
+  languages/mod_python
+  directories/mod_ldap
+  databases/mod_mariadb
+  applications/mod_signalwire
+  codecs/mod_bv
+  codecs/mod_ilbc
+  codecs/mod_silk
+  languages/mod_v8
+  applications/mod_avmd
+  event_handlers/mod_cdr_mongodb
+  applications/mod_cv
+  event_handlers/mod_erlang_event
+  event_handlers/mod_rayo
+  applications/mod_snom
+  event_handlers/mod_snmp
+  asr_tts/mod_pocketsphinx
+  endpoints/mod_verto
+  applications/mod_av
+  languages/mod_basic
+  event_handlers/mod_radius_cdr
+  endpoints/mod_rtmp
+  endpoints/mod_skinny
+)
 manual_pkgs=(
 freeswitch-all
 freeswitch
@@ -330,12 +355,12 @@ Build-Depends:
  libogg-dev, libspeex-dev, libspeexdsp-dev,
 # configure options
  libssl1.0-dev | libssl-dev, unixodbc-dev, libpq-dev,
- libncurses5-dev, libjpeg62-turbo-dev | libjpeg-turbo8-dev | libjpeg62-dev | libjpeg8-dev,
+ libncurses5-dev | libncurses-dev, libjpeg62-turbo-dev | libjpeg-turbo8-dev | libjpeg62-dev | libjpeg8-dev,
  python-dev | python-dev-is-python2 | python-dev-is-python3, python3-dev, python-all-dev | python3-all-dev, python-support (>= 0.90) | dh-python, erlang-dev, libtpl-dev (>= 1.5),
 # documentation
  doxygen,
 # for APR (not essential for build)
- uuid-dev, libexpat1-dev, libgdbm-dev, libdb-dev,
+ uuid-dev, libexpat1-dev, libgdbm-dev, libgdbm-compat-dev, libdb-dev,
 # used by many modules
  libcurl4-openssl-dev | libcurl4-gnutls-dev | libcurl-dev,
  bison, zlib1g-dev, libsofia-sip-ua-dev (>= 1.13.17),
diff --git a/debian/control-modules b/debian/control-modules
index 0f0784d3f5..830e43d83a 100644
--- a/debian/control-modules
+++ b/debian/control-modules
@@ -11,7 +11,7 @@ Description: Abstraction layer for APIs
 Module: applications/mod_av
 Description: mod_av
  Adds mod_av.
-Build-Depends: libavformat-dev, libswscale-dev, libavresample-dev
+Build-Depends: libavformat-dev, libswscale-dev, libavresample-dev | libswresample-dev
 Build-Depends-Bookworm: libavformat-dev, libswscale-dev, libswresample-dev
 
 Module: applications/mod_avmd
@@ -39,7 +39,7 @@ Description: Caller ID name lookup
 Module: applications/mod_cluechoo
 Description: Clue Choo
  This demo module renders a Clue Choo train on the FreeSWITCH console.
-Build-Depends: libncurses5-dev
+Build-Depends: libncurses5-dev | libncurses-dev
 
 Module: applications/mod_commands
 Description: Command module
@@ -661,7 +661,7 @@ Build-Depends-Bookworm: libperl-dev, libgdbm-compat-dev
 Module: languages/mod_python
 Description: mod_python
  Adds mod_python.
-Build-Depends: python-dev | python-dev-is-python2
+Build-Depends: python-dev | python-dev-is-python2 | python-dev-is-python3
 
 Module: languages/mod_python3
 Description: mod_python3
diff --git a/debian/rules b/debian/rules
index 88aaee3f71..2befc97bdd 100755
--- a/debian/rules
+++ b/debian/rules
@@ -74,7 +74,7 @@ override_dh_auto_clean:
 		--host=$(DEB_HOST_GNU_TYPE) --build=$(DEB_BUILD_GNU_TYPE) \
 		--prefix=/usr --localstatedir=/var --sysconfdir=/etc \
 		--with-gnu-ld --with-python --with-python3 --with-erlang --with-openssl \
-		--enable-core-odbc-support
+		--enable-core-odbc-support --disable-libvpx
 	touch $@
 
 override_dh_auto_configure: .stamp-configure
