diff --git a/debian/bootstrap.sh b/debian/bootstrap.sh
index 66adadd025..4cfceddce7 100755
--- a/debian/bootstrap.sh
+++ b/debian/bootstrap.sh
@@ -2,7 +2,7 @@
 ##### -*- mode:shell-script; indent-tabs-mode:nil; sh-basic-offset:2 -*-
 ##### Author: Travis Cross <tc@traviscross.com>
 
-codename="sid"
+codename="focal"
 modulelist_opt=""
 modules_add=""
 use_sysvinit=""
@@ -30,7 +30,7 @@ lang_dir="../conf/vanilla/lang"
 fs_description="FreeSWITCH is a scalable open source cross-platform telephony platform designed to route and interconnect popular communication protocols using audio, video, text or any other form of media."
 mod_build_depends="." mod_depends="." mod_recommends="." mod_suggests="."
 supported_debian_distros="wheezy jessie stretch buster sid"
-supported_ubuntu_distros="trusty utopic xenial"
+supported_ubuntu_distros="trusty utopic xenial focal"
 supported_distros="$supported_debian_distros $supported_ubuntu_distros"
 avoid_mods=(
   applications/mod_sms_flowroute
@@ -97,6 +97,23 @@ avoid_mods_xenial=(
   asr_tts/mod_flite
   loggers/mod_raven
 )
+avoid_mods_focal=(
+  directories/mod_ldap
+  databases/mod_mariadb
+  applications/mod_signalwire
+  codecs/mod_bv
+  codecs/mod_ilbc
+  codecs/mod_silk
+  languages/mod_v8
+  applications/mod_avmd
+  event_handlers/mod_cdr_mongodb
+  applications/mod_cv
+  event_handlers/mod_erlang_event
+  event_handlers/mod_rayo
+  languages/mod_perl
+  applications/mod_snom
+  event_handlers/mod_snmp
+)
 manual_pkgs=(
 freeswitch-all
 freeswitch
